### ISSUE TITLE
Feature Request for Replication/High Availability

### ISSUE THREAD

**Issue by ekrekeler** [nitnelave/lldap#126]
I think this is a great alternative to OpenLDAP and FreeIPA for self-hosted environments, with just one caveat.

Correct me if I am wrong, but it appears that lldap doesn't support any replication or high availability features if I want to run this on more than one server. I'm just opening an issue for it in case anyone else would be interested in such a thing.

**Comment by nitnelave** [comment]
Hmm, this is not something that I really had considered... Although it should be straightforward once we support other databases. I think once we support Postgresql, you should be able to put as many frontends as you want in front of the DB. High availability for the DB is left as an exercise to the reader :)

**Comment by ekrekeler** [comment]
I agree, it should be possible to leverage replication abilities of external databases once they are supported. Let me know if you would like help testing any of that.

**Comment by nitnelave** [comment]
I'll mark it as duplicate of https://github.com/nitnelave/lldap/issues/87

**Comment by nitnelave** [comment]
It might be better as a separate discussion, but LLDAP is made to be stateless, in order to have several instances running in parallel. As a result, an external program (e.g. another LLDAP) can modify the DB without warning during normal operation. 
The only "race" I can think of is one instance modifying an entry while another deletes it, in which case an error is the only logical response (but also, you have 2 admins doing weird contradicting stuff at the same time!)
Anything else is guarded by transactions and should work seamlessly.

**Comment by xxxserxxx** [comment]
@nitnelave Excellent! Then, assuming that the implementation follows the intention and is indeed stateless, replication should easy using external tools. The only caveat I can think of is that there should be only one master at a time between syncs, so we don't get internal DB ID conflicts... but, again, for simple cases such as fail-over and localization, this is perfect.

Thank you!
